pkg/kubectl/cmd/create.go

```diff
@@ -156,9 +157,23 @@ func RunCreate(f *cmdutil.Factory, cmd *cobra.Command, out io.Writer, options *C
 
 // createAndRefresh creates an object from input info and refreshes info with that object
 func createAndRefresh(info *resource.Info) error {
-	obj, err := resource.NewHelper(info.Client, info.Mapping).Create(info.Namespace, true, info.Object)
-	if err != nil {
-		return err
+	var obj runtime.Object
+	var err error
+	if info.Dynamic {
+		dc, err := info.DynamicClient()
```

Comment:
K - i'm happy to help dig in and fix anything that is broken - TPR ended up significantly complicating this code and I suspect all of your fixes are things that TPR should fall under.  I.e., unversioned objects, dynamic clients, codec hiding the details of the code.